hoot-core/src/main/cpp/hoot/core/io/OsmXmlChangesetFileWriter.cpp

```diff
@@ -231,8 +231,8 @@ void OsmXmlChangesetFileWriter::_writeNode(QXmlStreamWriter& writer, ConstNodePt
         it.key() != MetadataTags::HootHash())
     {
       writer.writeStartElement("tag");
-      writer.writeAttribute("k", _invalidCharacterRemover.removeInvalidCharacters(it.key()));
-      writer.writeAttribute("v", _invalidCharacterRemover.removeInvalidCharacters(it.value()));
+      writer.writeAttribute("k", _invalidCharacterEncoder.encodeInvalidCharacters(it.key()));
+      writer.writeAttribute("v", _invalidCharacterEncoder.encodeInvalidCharacters(it.value()));
       writer.writeEndElement();
     }
```

Comment:
The XML writer was removing the invalid chars rather than encoding them, and the reader wasn't handling them at all.  Made more sense to me to change to encode/decode them.

Reply:
There were some problems with the Djibouti data that needed the data removed.  I'll check and make sure that encoded data will still work.  It was a while ago that I chose to remove those invalid characters.  Hopefully encoding will work too because that is the better solution.

Reply:
Oh, ok.  No problem....I can change that part back, if needed.  What I mainly needed was the encoding on the reader, so I wasn't sending those chars to the translator in the name.